streaming/streaming.go

```diff
@@ -0,0 +1,78 @@
+package streaming
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/sul-dlss-labs/taco/config"
+)
+
+// Stream the interface for streaming pipeline
+type Stream interface {
+	SendMessage(string) error
+	// Do we want to expose these methods?
+	// createStream()
+	// waitUntilStreamExists()
+}
+
+type kinesisStream struct {
+	streamName *string
+	connection *kinesis.Kinesis
+}
+
+// NewKinesisStream create a new kinesis stream
+func NewKinesisStream(config *config.Config) (Stream, error) {
+	s := session.New(&aws.Config{Region: aws.String(config.AWSRegion),
+		Endpoint:   aws.String(config.KinesisEndpoint),
+		DisableSSL: aws.Bool(config.KinesisDisableSSL)})
+
+	kc := kinesis.New(s)
+
+	streamName := aws.String(config.DepositStreamName)
+
+	return &kinesisStream{streamName: streamName, connection: kc}, nil
+}
+
+func (d kinesisStream) SendMessage(message string) error {
+	// d.createStream()
+	// d.waitUntilStreamExists()
+
+	streams, err := d.connection.DescribeStream(&kinesis.DescribeStreamInput{StreamName: d.streamName})
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%v\n", streams)
+
+	putOutput, err := d.connection.PutRecord(&kinesis.PutRecordInput{
+		Data:         []byte(message),
+		StreamName:   d.streamName,
+		PartitionKey: aws.String("key1"),
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%v\n", putOutput)
+	return nil
+}
+
+func (d kinesisStream) createStream() error {
```

Comment:
I thought we said yesterday that we were not going to create a stream in the app but just expect that the stream it will publish messages to already exists

Reply:
This is a capability we have, but we're not calling it right now. See line 39 in the same file.

Reply:
so if we are not exercising them, should we remove them? FWIW the role that the containers will have will not allow for the app to create a stream so the lines would not work even if exercised.

Reply:
Okay, I can remove.

Reply:
It has been removed.